setup.py

```diff
@@ -87,141 +64,105 @@ def cmake_version_check():
+        # extensions as a one-shot cmake invocation. Usually we'd build each
+        # separate using build_extension.
         self.build()
 
     def build(self):
-        _ctx.ext_dir = os.path.abspath(self.build_lib)
-
-        _ctx.build_temp_dir = (
+        self.build_temp_dir = (
             os.environ.get("OTIO_CXX_BUILD_TMP_DIR")
             or os.path.abspath(self.build_temp)
         )
-        _ctx.debug = self.debug or bool(os.environ.get("OTIO_CXX_DEBUG_BUILD"))
 
-        if not _ctx.ext_dir.endswith(os.path.sep):
-            _ctx.ext_dir += os.path.sep
-        if not os.path.exists(_ctx.build_temp_dir):
-            os.makedirs(_ctx.build_temp_dir)
+        if not os.path.exists(self.build_temp_dir):
+            os.makedirs(self.build_temp_dir)
 
-        cmake_generate()
-        cmake_install('Debug' if _ctx.debug else 'Release')
+        debug = (self.debug or bool(os.environ.get("OTIO_CXX_DEBUG_BUILD")))
+        self.build_config = ('Debug' if debug else 'Release')
+
+        self.cmake_generate()
+        self.cmake_install()
+
+    def cmake_generate(self):
+        # Determine the target directory for the built library
+        install_dir = os.path.abspath(self.build_lib)
+        if not install_dir.endswith(os.path.sep):
+            install_dir += os.path.sep
+
+        cmake_args = [
+            '-DPYTHON_EXECUTABLE=' + sys.executable,
+            '-DOTIO_PYTHON_INSTALL:BOOL=ON',
+            '-DOTIO_CXX_INSTALL:BOOL=ON',
+            '-DCMAKE_BUILD_TYPE=' + self.build_config,
+        ]
+
+        # install the C++ into the opentimelineio/cxx-sdk directory under the
+        # python installation
+        cmake_install_prefix = os.path.join(
+            install_dir,
+            "opentimelineio",
+            "cxx-sdk"
+        )
+
+        cmake_args += [
+            '-DOTIO_PYTHON_INSTALL_DIR=' + install_dir,
+            '-DCMAKE_INSTALL_PREFIX=' + cmake_install_prefix,
+        ]
+
+        if platform.system() == "Windows":
+            if sys.maxsize > 2**32:
+                cmake_args += ['-A', 'x64']
+
+        cxx_coverage = bool(os.environ.get("OTIO_CXX_COVERAGE_BUILD"))
+        if cxx_coverage and not os.environ.get("OTIO_CXX_BUILD_TMP_DIR"):
+            raise RuntimeError(
+                "C++ code coverage requires that both OTIO_CXX_COVERAGE_BUILD=ON "
+                "and OTIO_CXX_BUILD_TMP_DIR are specified as environment "
+                "variables, otherwise coverage cannot be generated."
+            )
+
+        if cxx_coverage:
+            cmake_args += ['-DOTIO_CXX_COVERAGE=1']
+
+        cmake_args = ['cmake', SOURCE_DIR] + cmake_args
+        subprocess.check_call(
+            cmake_args,
+            cwd=self.build_temp_dir,
+            env=os.environ.copy()
+        )
+
+    def cmake_install(self):
+        if platform.system() == "Windows":
+            multi_proc = '/m'
+        else:
+            multi_proc = '-j2'
```

Comment:
conservative :)

Reply:
Hah, I just relocated that line from the old setup. Probably worth re-evaluating at that :)

Reply:
if we want to detect the number of cores:

`multiprocessing.cpu_count()`